src/server/services/hli/stadspas-router-external-consumer.ts

```diff
@@ -0,0 +1,177 @@
+    message: 'Onbekende error',
+  },
+} as const;
+
+async function sendAdministratienummerResponse(
+  req: Request<{ token: string }>,
+  res: Response
+) {
+  let authProfileAndToken: AuthProfileAndToken | null = null;
+  let error: ApiError = errors.UNKNOWN;
+
+  try {
+    authProfileAndToken = await getAuth(req);
+  } catch (error) {
+    error = errors.DIGID_AUTH;
+  }
+
+  if (
+    authProfileAndToken?.profile.id &&
+    authProfileAndToken.profile.profileType === 'private'
+  ) {
+    const administratienummerResponse = await fetchAdministratienummer(
+      res.locals.requestID,
+      authProfileAndToken
+    );
+
+    // Administratienummer found, encrypt and sent
+    if (
+      administratienummerResponse.status === 'OK' &&
+      administratienummerResponse.content
+    ) {
+      const [administratienummerEncrypted] =
+        administratienummerResponse.content !== null
+          ? encrypt(administratienummerResponse.content)
+          : [];
+
+      const requestConfig = getApiConfig('AMSAPP', {
+        data: {
+          administratienummerEncrypted,
+          token: req.params.token,
+        },
+      });
+
+      // Deliver the token with administratienummer to app.amsterdam.nl
+      const deliveryResponse = await requestData(
+        requestConfig,
+        res.locals.requestID
+      );
+
+      if (deliveryResponse.status === 'OK') {
+        // Send app protocol header, should open app
+        return res.redirect(AMSAPP_STADSPAS_DEEP_LINK);
+      }
+
+      if (deliveryResponse.status === 'ERROR') {
+        // Delivery response error
+        error = errors.AMSAPP_DELIVERY_FAILED;
+      }
+    }
+
+    // administratienummer not found in Zorgned
+    if (!administratienummerResponse.content) {
+      error = errors.ADMINISTRATIENUMMER_NOT_FOUND;
+    }
+
+    // administratienummer error Response
+    if (administratienummerResponse.status === 'ERROR') {
+      error = errors.ADMINISTRATIENUMMER_RESPONSE_ERROR;
+    }
+
+    if (error) {
+      return res.render('amsapp-stadspas-administratienummer', {
+        error: error,
+        appHref: `${AMSAPP_STADSPAS_DEEP_LINK}?errorMessage=${error.message}&errorCode=${error.code}`,
+      });
+    }
+  }
+
+  return sendUnauthorized(res);
+}
+
+router.get(
+  BffEndpoints.STADSPAS_ADMINISTRATIENUMMER,
+  sendAdministratienummerResponse
+);
+
+function apiKeyHandler(req: Request, res: Response, next: NextFunction) {
+  // TODO: implement api key validation
+  next();
+}
+
+router.get(
+  BffEndpoints.STADSPAS_PASSEN,
+  apiKeyHandler,
+  (
+    req: Request<{ administratienummerEncrypted: string }>,
+    res: Response,
+    next: NextFunction
+  ) => {
+    let reason = '';
+    try {
+      const administratienummerEncrypted =
+        req.params.administratienummerEncrypted;
+
+      if (administratienummerEncrypted) {
+        const clientnummer = decrypt(administratienummerEncrypted);
+        // TODO: Implement service call with clientnummer param
```

Comment:
is dat nog niet gebeurt?

Reply:
Nee, dat gebeurt in https://github.com/Amsterdam/mijn-amsterdam-frontend/pull/1372